outbox/order-service/src/main/java/io/debezium/examples/outbox/order/outbox/EventSender.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.examples.outbox.order.outbox;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.enterprise.event.TransactionPhase;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import io.debezium.examples.outbox.order.event.ExportedEvent;
+
+@ApplicationScoped
+public class EventSender {
+
+    @Inject
+    private Event<OutboxEventSubmitted> submitted;
+
+    @Inject
+    private HouseKeeper houseKeeper;
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    public void onExportedEvent(@Observes ExportedEvent event) {
+        OutboxEvent outboxEvent = new OutboxEvent(
+                event.getAggregateType(), event.getAggregateId(), event.getType(), event.getPayload()
+        );
+        entityManager.persist(outboxEvent);
+
+        submitted.fire(new OutboxEventSubmitted(outboxEvent.getId()));
+    }
+
+    public void onEventSubmitted(@Observes(during=TransactionPhase.AFTER_COMPLETION) OutboxEventSubmitted event) {
```

Comment:
And remove the entry right away. That way the event table is always empty, we'll get the new events from the log.